cedar-policy-core/src/parser.rs

```diff
@@ -56,17 +56,21 @@ pub fn parse_policyset_to_ests_and_pset(
     let mut errs = Vec::new();
     let cst = text_to_cst::parse_policies(text).map_err(err::ParseErrors)?;
     let pset = cst.to_policyset(&mut errs);
-    let ests = cst
-        .with_generated_policyids()
-        .expect("shouldn't be None since parse_policies() didn't return Err")
-        .map(|(id, policy)| match &policy.node {
-            Some(p) => Ok(Some((id, p.clone().try_into()?))),
-            None => Ok(None),
-        })
-        .collect::<Result<Option<HashMap<ast::PolicyID, est::Policy>>, err::ParseErrors>>()?;
-    match (errs.is_empty(), ests, pset) {
-        (true, Some(ests), Some(pset)) => Ok((ests, pset)),
-        (_, _, _) => Err(err::ParseErrors(errs)),
+    if let Some(pset) = pset {
+        let ests = cst
+            .with_generated_policyids()
+            .expect("shouldn't be None since parse_policies() didn't return Err")
```

Comment:
Nit: update `expect` message to say that neither `parse_policies` nor `to_policyset` returned an error.